scalafix/rules/src/main/scala/rsc/rules/pretty/SemanticdbPrinter.scala

```diff
@@ -130,9 +156,10 @@ class SemanticdbPrinter(env: Env, index: DocumentIndex, config: RscCompatConfig)
             rep("[", args, ", ", "]")(normal)
           }
         case s.SingleType(pre, sym) =>
-          // TODO: Also check for config.better.
-          val prettyPre = if (pre == s.NoType) sym.trivialPrefix(env) else pre
-          opt(prettyPre, ".")(prefix)
+          if (!config.better) {
```

Comment:
This logic is incorrect because it doesn't check whether `sym` is in scope. If it isn't, the prefix cannot be omitted.

Reply:
I originally checked if sym was in scope but it ended up printing one extra layer of name, as in collection.mutable.map instead of mutable.map. Will need to investigate further it seems.